xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -112,7 +104,14 @@ void CMusicInfoScanner::Process()
         m_handle = dialog->GetHandle(g_localizeStrings.Get(314));
     }
 
-    m_bClean = g_advancedSettings.m_bMusicLibraryCleanOnUpdate;
+    // check if we only need to perform a cleaning
+    if (m_bClean && m_pathsToScan.empty())
+    {
+      CleanDatabase(false);
+      m_bRunning = false;
+
+      return;
```

Comment:
Sorry I don't see any "side effects"? Getting "tick" I guess? The db does need to be open for just cleaning as well as the scanning, although it will open it if it isn't. I can move those, but we do want the progress display when not being done background in case delete takes noticeable time, so branch is not at beginning.